core/domain/stats_jobs_one_off.py

```diff
@@ -1373,3 +1373,52 @@ def reduce(key, items):
             yield (key, len(items))
         else:
             yield (key, items)
+
+
+class ExplorationMissingStatsAudit(jobs.BaseMapReduceOneOffJobManager):
+    """A one-off job for finding explorations that are missing stats models.
+
+    If this job outputs any explorations, then the following two jobs need to be
+    run in succession:
+        - RegenerateMissingV1StatsModelsOneOffJob
+        - RegenerateMissingV2StatsModelsOneOffJob
+
+    Otherwise, an empty output signals that all explorations have stats.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel, stats_models.ExplorationStatsModel]
```

Comment:
Seems like a weird way to do it. Why not just iterate over ExplorationModel only, ignore deleted ones, and then check if the stats model exists / is deleted? Usually we iterate over just one model type.

(Also, for the more comprehensive model relation checks -- e.g. every ExpStatsModel must have a corresponding ExpModel -- please make sure that prod_validation_one_off_jobs is up to date.)

Reply:
> Seems like a weird way to do it. Why not just iterate over ExplorationModel only, ignore deleted ones, and then check if the stats model exists / is deleted? Usually we iterate over just one model type.

I decided to iterate both since it seems valuable to see whether we have any "zombie stats" that refer to deleted explorations. Also, `StatisticsAudit` in this same file also iterates over two items so I think it's OK.

> Also, for the more comprehensive model relation checks -- e.g. every ExpStatsModel must have a corresponding ExpModel -- please make sure that prod_validation_one_off_jobs is up to date.
Sorry, not clear to me what's being asked for here. Are you asking for this job to become a `ProdValidationOneOffJob` sub-class?

Reply:
@ankita240796 for advice here

Reply:
I suggest that:
- For this one-off job, just check whether each exploration has valid stats. No need to check anything else.
- In general we encode data validity info in our prod validation one-off jobs. The constraints you are mentioning here need to be part of those jobs, so that when we run audits in the future, any violations are caught.

Does this clarify?

Reply:
Yup, when looking at the exploration model logic it looks like it's "allowed" for explorations to have no stats models. When this happens, a new empty model is created to compensate: https://github.com/oppia/oppia/blob/37e46f79ac42c0b8fd5904addd06fc9767553f7e/core/domain/exp_services.py#L572-L577

It's not clear to me when this would fail, however.

Reply:
OK, thanks for the investigation! I think the recommendations/thoughts above still apply, we can do a spot-check once we get the results to see if they all come from old explorations.